pandas/tests/io/excel/test_writers.py

```diff
@@ -256,26 +272,10 @@ class and any subclasses, on account of the `autouse=True`
         option_name = "io.excel.{ext}.writer".format(ext=ext.strip("."))
         prev_engine = get_option(option_name)
         set_option(option_name, engine)
-        with ensure_clean(ext) as path:
-            self.path = path
-            yield
+        yield
         set_option(option_name, prev_engine)  # Roll back option change
```

Comment:
as a follow-on could just use `with option_context`, see https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.option_context.html

Reply:
for consistency with the reader tests as a follow-on this fixture could instead set engine explicitly using a partial

https://github.com/pandas-dev/pandas/blob/04e67c46e5c9f93f26d41d9e970dc7554e80916c/pandas/tests/io/excel/test_readers.py#L83

and 

https://github.com/pandas-dev/pandas/blob/8ea102acdb45bb70cb30ea77108a50054c28c24d/pandas/tests/io/excel/test_odf.py#L12-L15

and eventually look to a single reusable fixture to consolidate these.

Reply:
@simonjayhawkins Should I make these changes as a separate PR after this current PR?

Reply:
They are suggestions for follow-ups. no action needed for this PR.